allennlp/training/metrics/boolean_accuracy.py

```diff
@@ -95,9 +101,13 @@ def get_metric(self, reset: bool = False):
             accuracy = float(self._correct_count) / float(self._total_count)
         else:
             accuracy = 0.0
+        if world_size > 1:
+            accuracy_tensor = torch.tensor(accuracy).to(cuda_device)
+            dist.all_reduce(accuracy_tensor, op=dist.ReduceOp.SUM)
+            accuracy = accuracy_tensor.item() / world_size
```

Comment:
This is only correct if each worker has the same `self._total_count`, right? Can we rely on that?